### PR TITLE
014 unit test

### PR DESCRIPTION
**Pull request by endrjux**
Unit Tests for vcdMaker

- catch framework as submodule
- CMake target `check` for compiling and running UTs
- Some stubs
- UTs for few classes (more is needed)

See #14.

### CODE REVIEW

File: sources/common/test/unitTest/EventSignal.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    SIGNAL::EventSignal signal2 = getEventSignal();
+    SIGNAL::Signal &rSignal2 = signal2;
+
+    DummySignal dummySignal(DummySignal::DUMMY_NAME,
+                            DummySignal::DUMMY_TIMESTAMP);
+    SIGNAL::Signal &rDummySignal = dummySignal;
+
+    SECTION("operator==")
+    {
+        REQUIRE_FALSE(rSignal1 == rSignal1);
```
Comment (WojciechRynczuk) [inline sources/common/test/unitTest/EventSignal.cpp]: General remark: I'm wondering if there's any particular reason for using references here and not just objects? It works as well and seems to be more straight forward.

Reference are OK in some cases, e.g. when verifying the ltComparatpr.

**Comment by WojciechRynczuk** [comment]
General remark: Maybe the unit testing files could be suffixed so as we wouldn't have two Signal.cpp files. What do you think about it?

File: sources/common/test/unitTest/UtMain.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/// @file UtMain.cpp
```
Comment (WojciechRynczuk) [inline sources/common/test/unitTest/UtMain.cpp]: Why do we need this file?

**Comment by WojciechRynczuk** [comment]
General remark: There are doxygen warnings that shall be fixed. Most of them shall be removed if you'd rebase to the v3_release branch. But still a few of them will remain and need to be addressed.

**Comment by WojciechRynczuk** [comment]
I was also thinking about making the process of running UT under Windows easier. But it seems to me not worth the effort at the moment. They can be run on Linux smoothly. I like the output :)

**Comment by WojciechRynczuk** [comment]
How about vcdMerge or vcdMaker specific tests? Will we keep them here as well?

File: sources/common/test/unitTest/EventSignal.cpp (hunk 1 above)
Comment (endrjux) [inline sources/common/test/unitTest/EventSignal.cpp]: The reason is that `EqualTo` is `virtual`. I just wanted to use references to ensure everything works ok.

File: sources/common/test/unitTest/UtMain.cpp (hunk 2 above)
Comment (endrjux) [inline sources/common/test/unitTest/UtMain.cpp]: This is required. This file contains `main()` for unit test.

**Comment by endrjux** [comment]
Some answers:
1. Directory structure is "prefix".
2. Will fix.
3. I think it should just work, but maybe something special needs to be done in CMake for VS.
4. They will be places in `\sources\<app>\test\unitTest`.

**Pull request merged**